prototyping/Luau/RuntimeError.agda

```diff
@@ -3,13 +3,14 @@ module Luau.RuntimeError where
 open import Agda.Builtin.Equality using (_≡_)
 open import Luau.Heap using (Heap; _[_])
 open import FFI.Data.Maybe using (just; nothing)
+open import FFI.Data.String using (String)
 open import Luau.Syntax using (Block; Expr; nil; var; addr; block_is_end; _$_; local_←_; return; done; _∙_)
 
 data RuntimeErrorᴮ {a} (H : Heap a) : Block a → Set
 data RuntimeErrorᴱ {a} (H : Heap a) : Expr a → Set
 
 data RuntimeErrorᴱ H where
-  NilIsNotAFunction : ∀ {M} → RuntimeErrorᴱ H (nil $ M)
+  ValueNotCallable : ∀ x → RuntimeErrorᴱ H x
```

Comment:
As a style point, I think we should have separate runtime errors for different kinds of non-functions.
```suggestion
  NilIsNotAFunction : ∀ {M} → RuntimeErrorᴱ H (nil $ M)
  NumberIsNotAFunction : ∀ {n M} → RuntimeErrorᴱ H (number n $ M)
```